Patch: 00_stolen_from_HEAD.patch
Author: Tollef Fog Heen <tfheen@debian.org>
Handle empty queue files.
Index: mailman-2.1.10/Mailman/Handlers/Scrubber.py
===================================================================
--- mailman-2.1.10.orig/Mailman/Handlers/Scrubber.py	2008-04-21 20:28:22.000000000 +0200
+++ mailman-2.1.10/Mailman/Handlers/Scrubber.py	2008-06-15 18:49:38.000000000 +0200
@@ -385,6 +385,8 @@
                     t = unicode(t, 'ascii', 'replace')
                 try:
                     # Should use HTML-Escape, or try generalizing to UTF-8
+                    if len(charset) == 0:
+                        charset = 'us-ascii'
                     t = t.encode(charset, 'replace')
                 except (UnicodeError, LookupError, ValueError,
                         AssertionError):
