Patch: 15_mailmanctl_daemonize.patch
Author: Tollef Fog Heen <tfheen@debian.org>
Daemonize mailmanctl properly (Closes: #175175)
Index: mailman-2.1.10/bin/mailmanctl
===================================================================
--- mailman-2.1.10.orig/bin/mailmanctl	2008-04-21 20:28:22.000000000 +0200
+++ mailman-2.1.10/bin/mailmanctl	2008-06-15 18:49:42.000000000 +0200
@@ -417,6 +417,13 @@
         # won't be opening any terminal devices, don't do the ultra-paranoid
         # suggestion of doing a second fork after the setsid() call.
         os.setsid()
+
+        # Be sure to close any open std{in,out,err}
+        devnull = os.open('/dev/null', 0)
+        os.dup2(devnull, 0)
+        os.dup2(devnull, 1)
+        os.dup2(devnull, 2)
+
         # Instead of cd'ing to root, cd to the Mailman installation home
         os.chdir(mm_cfg.PREFIX)
         # Set our file mode creation umask
