Patch: 32_MIME_fixup.patch
Author: Tollef Fog Heen <tfheen@debian.org>
Handle empty queue files (Closes: #223906)
Index: mailman-2.1.10/Mailman/Handlers/Decorate.py
===================================================================
--- mailman-2.1.10.orig/Mailman/Handlers/Decorate.py	2008-04-21 20:28:22.000000000 +0200
+++ mailman-2.1.10/Mailman/Handlers/Decorate.py	2008-06-15 18:49:45.000000000 +0200
@@ -191,6 +191,7 @@
     del msg['content-transfer-encoding']
     del msg['content-disposition']
     msg['Content-Type'] = 'multipart/mixed'
+    msg['Mime-version'] = '1.0'
 
 
 
