Patch: 78_DeprecationWarning.patch
Author: Lionel Elie Mamane <lionel@mamane.lu>
Shut up DeprecationWarning in Python 2.4 (Closes: #341699)
Index: mailman-2.1.12/Mailman/Handlers/Scrubber.py
===================================================================
--- mailman-2.1.12.orig/Mailman/Handlers/Scrubber.py	2009-03-02 20:50:20.000000000 +0100
+++ mailman-2.1.12/Mailman/Handlers/Scrubber.py	2009-03-02 20:50:24.000000000 +0100
@@ -25,6 +25,7 @@
 import errno
 import binascii
 import tempfile
+import warnings
 from cStringIO import StringIO
 from types import IntType, StringType
 
@@ -77,6 +78,10 @@
             check(mimetypes.common_types)
         return all
 
+# FIXME: Remove for Mailman 2.2 (which will drop support for Python 2.1)
+# and then actually replace part.get_type(part.get_default_type())
+# with part.get_content_type()
+warnings.filterwarnings("ignore","^get_type\(\) deprecated;",DeprecationWarning,"Mailman")
 
 
 def guess_extension(ctype, ext):
