diff -upr mailman-2.1.16rc2/Mailman/Defaults.py.in mailman-2.1.16rc2-ja-utf8/Mailman/Defaults.py.in
--- mailman-2.1.16rc2/Mailman/Defaults.py.in	2013-07-20 08:32:35.000000000 +0900
+++ mailman-2.1.16rc2-ja-utf8/Mailman/Defaults.py.in	2013-09-20 01:17:44.643002441 +0900
@@ -1548,7 +1548,7 @@ add_language('hr',    _('Croatian'),
 add_language('hu',    _('Hungarian'),           'iso-8859-2',  'ltr')
 add_language('ia',    _('Interlingua'),         'iso-8859-15', 'ltr')
 add_language('it',    _('Italian'),             'iso-8859-1',  'ltr')
-add_language('ja',    _('Japanese'),            'euc-jp',      'ltr')
+add_language('ja',    _('Japanese'),            'utf-8',       'ltr')
 add_language('ko',    _('Korean'),              'euc-kr',      'ltr')
 add_language('lt',    _('Lithuanian'),          'iso-8859-13', 'ltr')
 add_language('nl',    _('Dutch'),               'iso-8859-1',  'ltr')
